### ISSUE TITLE
Work out detailed rules for scopes and write a blog post about it

### ISSUE THREAD

**Issue by gvanrossum** [faster-cpython/ideas#249]
I've got the rules worked out, and some text and some Python code that describes them. I'd like to finish this up in the form of a blog post hosted on Medium (or elsewhere if Medium proves to be a problem).

**Comment by gramster** [comment]
I personally am against the use of Medium; is there somewhere else you could post this? You could of course use our Python blog but I expect you'd prefer somewhere more neutral.

**Comment by gvanrossum** [comment]
Yeah, I just tried Medium after a long absence and found that it has turned into a junkyard. My current plan is to publish using my GitHub home page. I write in MarkDown and GitHub publishes decent HTML, which makes me happy.

**Comment by gvanrossum** [comment]
The blog will appear at https://gvanrossum.github.io/formal/scopesblog.html. (What's there now is incomplete.)

**Comment by gvanrossum** [comment]
I consider the text (and code) at https://gvanrossum.github.io/formal/scopesblog.html now basically complete. Have a look!

**Comment by iritkatriel** [comment]
You say LEGB, but the code of ClosedScope.lookup seems to be doing LGE(B)?

**Comment by gvanrossum** [comment]
@iritkatriel 
> You say LEGB, but the code of ClosedScope.lookup seems to be doing LGE(B)?

`name in self.globals` is true iff there's a `global <name>` declaration in the scope.

**Comment by gvanrossum** [comment]
I consider this done. On to code generation and execution.

**Comment by oraluben** [comment]
@gvanrossum Thanks for the blog! I've found a trivial markdown syntax issue near "hoops" at https://github.com/gvanrossum/gvanrossum.github.io/blob/876b4b89a3b715a05e95165135556e79f9c3db55/formal/scopesblog.md?plain=1#L53
